sdcflows/workflows/tests/test_pepolar.py

```diff
@@ -0,0 +1,194 @@
+                         desc=None, extension=['.nii.gz', '.nii'])
+
+    with pytest.raises(ValueError):
+        check_pes([
+            (im.path, im.get_metadata()['PhaseEncodingDirection'])
+            for im in epidata], bold.get_metadata()['PhaseEncodingDirection'])
+
+
+def test_prepare_epi_wf1(bids_layouts, tmpdir):
+    """Test preparation workflow."""
+    tmpdir.chdir()
+
+    layout = bids_layouts['testdata']
+
+    bold = layout.get(suffix='bold', dir='LR', direction='LR',
+                      extension=['.nii.gz', '.nii'])[0]
+    boldref = layout.get(suffix='boldref', dir='LR', direction='LR',
+                         extension=['.nii.gz', '.nii'])[0]
+    epidata = layout.get(suffix='epi', desc=None, extension=['.nii.gz', '.nii'])
+
+    matched_pe = check_pes([(im.path, im.get_metadata()['PhaseEncodingDirection'])
+                            for im in epidata], bold.get_metadata()['PhaseEncodingDirection'])
+
+    assert matched_pe is True
+
+    wf = init_prepare_epi_wf(omp_nthreads=1, matched_pe=matched_pe)
+    wf.inputs.inputnode.maps_pe = [(im.path, im.get_metadata()['PhaseEncodingDirection'])
+                                   for im in epidata]
+    wf.inputs.inputnode.ref_brain = boldref.path
+    wf.inputs.inputnode.epi_pe = bold.get_metadata()['PhaseEncodingDirection']
+
+
+def test_prepare_epi_wf2(bids_layouts, tmpdir):
+    """Test preparation workflow."""
+    tmpdir.chdir()
+
+    layout = bids_layouts['testdata']
+
+    bold = layout.get(suffix='bold', dir='LR', direction='LR',
+                      extension=['.nii.gz', '.nii'])[0]
+    boldref = layout.get(suffix='boldref', dir='LR', direction='LR',
+                         extension=['.nii.gz', '.nii'])[0]
+    epidata = layout.get(suffix='epi', dir='RL', direction='RL',
+                         desc=None, extension=['.nii.gz', '.nii'])
+
+    matched_pe = check_pes([(im.path, im.get_metadata()['PhaseEncodingDirection'])
+                            for im in epidata], bold.get_metadata()['PhaseEncodingDirection'])
+
+    assert matched_pe is False
+
+    wf = init_prepare_epi_wf(omp_nthreads=1, matched_pe=matched_pe)
+    wf.inputs.inputnode.maps_pe = [(im.path, im.get_metadata()['PhaseEncodingDirection'])
+                                   for im in epidata]
+    wf.inputs.inputnode.ref_brain = boldref.path
+    wf.inputs.inputnode.epi_pe = bold.get_metadata()['PhaseEncodingDirection']
+
+
+@pytest.mark.parametrize('dataset', [
+    'ds001600',
+    'testdata',
+])
+def test_pepolar_wf1(bids_layouts, output_path, dataset, workdir):
+    """Test preparation workflow."""
+    layout = bids_layouts[dataset]
+
+    if dataset == 'testdata':
+        bold = layout.get(suffix='bold', dir='LR', direction='LR',
+                          extension=['.nii.gz', '.nii'])[0]
+        boldref = layout.get(suffix='boldref', dir='LR', direction='LR', desc='brain',
+                             extension=['.nii.gz', '.nii'])[0]
+    elif dataset == 'ds001600':
```

Comment:
Does this work? There are lots of bold datasets in here, each with a different fieldmap